modules/serverpod_auth/serverpod_auth_server/test/firebase/firebase_auth_mock.dart

```diff
@@ -0,0 +1,189 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:jose/jose.dart';
+
+class MockAuthBaseClient extends http.BaseClient {
```

Comment:
Question: Have you seen the MockClient class that is provided by the `http` package? Could that be used instead of subclassing?

Source here: https://pub.dev/documentation/http/1.2.1/http.testing/MockClient-class.html